samples/javascript_nodejs/49.qnamaker-all-features/dialogs/qnamakerBaseDialog.js

```diff
@@ -27,10 +31,11 @@ class QnAMakerBaseDialog extends QnAMakerDialog {
      * Core logic of QnA Maker dialog.
      * @param {QnAMaker} qnaService A QnAMaker service object.
      */
-    constructor(knowledgebaseId, authkey, host) {
-        var noAnswer = ActivityFactory.fromObject(DefaultNoAnswer);
+    constructor(knowledgebaseId, authkey, host, defaultAnswer) {
+
+        var defaultAnswerActivity = MessageFactory.text(!!defaultAnswer.trim() ? defaultAnswer : DefaultAnswer);
```

Comment:
nit but `var` should almost always be either `let` or `const` depending on whether the variable is allowed to be reassigned.

Reply:
done. thanks!